Subscription None in GroupCoordinator._do_group_rejoin

Hi,
thanks for the hard work on version `0.4.0`. I'm watching updates on this repo and it is pleasure to see how new features and issues are solved. 
I'm trying to update to version `0.4.0` (using kafka `0.9.0.1`, python 3.5.2) and I've encountered strange error for only one test of our wrapper over `aiokafka`. When I'm stopping consumer and unsubscribing from a topic I get following error:
```python
messaging_client/mqclient/kafka_consumer.py:110: in stop
    await self._consumer.stop()
../../.virtualenvs/messaging/lib/python3.5/site-packages/aiokafka/consumer/consumer.py:340: in stop
    yield from self._coordinator.close()
../../.virtualenvs/messaging/lib/python3.5/site-packages/aiokafka/consumer/group_coordinator.py:273: in close
    yield from self._coordination_task
/usr/lib/python3.5/asyncio/futures.py:361: in __iter__
    yield self  # This tells Task to wait for completion.
/usr/lib/python3.5/asyncio/tasks.py:296: in _wakeup
    future.result()
/usr/lib/python3.5/asyncio/futures.py:274: in result
    raise self._exception
../../.virtualenvs/messaging/lib/python3.5/site-packages/aiokafka/consumer/group_coordinator.py:221: in _on_coordination_done
    fut.result()
/usr/lib/python3.5/asyncio/futures.py:274: in result
    raise self._exception
/usr/lib/python3.5/asyncio/tasks.py:239: in _step
    result = coro.send(None)
../../.virtualenvs/messaging/lib/python3.5/site-packages/aiokafka/consumer/group_coordinator.py:559: in _coordination_routine
    success = yield from self._do_rejoin_group(subscription)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <aiokafka.consumer.group_coordinator.GroupCoordinator object at 0x7f484fcab8d0>, subscription = <aiokafka.consumer.subscription_state.Subscription object at 0x7f484f112470>

    @asyncio.coroutine
    def _do_rejoin_group(self, subscription):
        rebalance = CoordinatorGroupRebalance(
            self, self.group_id, self.coordinator_id,
            subscription, self._assignors, self._session_timeout_ms,
            self._retry_backoff_ms, loop=self._loop)
        assignment = yield from rebalance.perform_group_join()
    
        if not subscription.active:
            log.debug("Subscription changed during rebalance from %s to %s. "
                      "Rejoining group.",
                      subscription.topics,
>                     self._subscription.subscription.topics)
E           AttributeError: 'NoneType' object has no attribute 'topics'

../../.virtualenvs/messaging/lib/python3.5/site-packages/aiokafka/consumer/group_coordinator.py:758: AttributeError
```
This is only happening in one of my test and unfortunately I am not able to extract it to self containing example. But we can see that `self._subscription.subscription` is `None` and this can easily happen when somebody call `client.unsubscribe` while awaiting something in `GroupCoordinator._coordinator_routine`, right? In `GroupCoordinator._coordinator_routine` is saved only reference to `Subscription` but if it is rewritten in `SubscriptionState` by None (by method `unsubscribe`) this reference suddenly points to something that is not used anymore.
This error is happening reliably, but when I change timing (add some `asyncio.sleep` before `consumer.stop`) it disappears. Still I think this can show to some problem with timing in coordinator. Am I on the right track?

--- a/aiokafka/consumer/consumer.py
+++ b/aiokafka/consumer/consumer.py
@@ -896,7 +896,7 @@ class AIOKafkaConsumer(object):
         Returns:
             frozenset: {topic, ...}
         """
-        return self._subscription.subscription.topics
+        return self._subscription.topics
 
     def unsubscribe(self):
         """ Unsubscribe from all topics and clear all assigned partitions. """

--- a/aiokafka/consumer/group_coordinator.py
+++ b/aiokafka/consumer/group_coordinator.py
@@ -755,7 +755,7 @@ class GroupCoordinator(BaseCoordinator):
             log.debug("Subscription changed during rebalance from %s to %s. "
                       "Rejoining group.",
                       subscription.topics,
-                      self._subscription.subscription.topics)
+                      self._subscription.topics)
             return False
         if assignment is None:
             return False

--- a/aiokafka/consumer/subscription_state.py
+++ b/aiokafka/consumer/subscription_state.py
@@ -54,6 +54,12 @@ class SubscriptionState:
     def listener(self) -> ConsumerRebalanceListener:
         return self._listener
 
+    @property
+    def topics(self):
+        if self._subscription is not None:
+            return self._subscription.topics
+        return set([])
+
     def assigned_partitions(self) -> Set[TopicPartition]:
         if self._subscription is None:
             return set([])
@@ -211,7 +217,8 @@ class SubscriptionState:
 
         Caller: Coordinator
         """
-        self._subscription._begin_reassignment()
+        if self._subscription is not None:
+            self._subscription._begin_reassignment()
 
     # Fetcher callable API:
 

--- a/tests/test_consumer.py
+++ b/tests/test_consumer.py
@@ -1460,3 +1460,17 @@ class TestConsumerIntegration(KafkaIntegrationTestCase):
             coordinator._exclude_internal_topics, exclude_internal_topics)
         self.assertEqual(
             coordinator._enable_auto_commit, enable_auto_commit)
+
+    @run_until_complete
+    def test_consumer_fast_unsubscribe(self):
+        # Unsubscribe before coordination finishes
+        consumer = AIOKafkaConsumer(
+            loop=self.loop, group_id="test_consumer_fast_unsubscribe",
+            bootstrap_servers=self.hosts)
+        tp = TopicPartition(self.topic, 0)
+
+        yield from consumer.start()
+        consumer.subscribe([tp])
+        yield from asyncio.sleep(0.01, loop=self.loop)
+        consumer.unsubscribe()
+        yield from consumer.stop()

--- a/tests/test_subscription_state.py
+++ b/tests/test_subscription_state.py
@@ -175,3 +175,11 @@ def test_assigned_state(subscription_state):
 
     assert repr(tp_state) == \
         "TopicPartitionState<Status=PartitionStatus.ASSIGNED position=None>"
+
+
+def test_begin_reassignment(subscription_state):
+    subscription_state.subscribe({"tp1", "tp2"})
+    subscription_state.unsubscribe()
+
+    # After unsubscribe we should not fail begin_reassignment, just ignore it
+    subscription_state.begin_reassignment()

